test/integration/template_collector_test.go

```diff
@@ -0,0 +1,240 @@
+		helmChartPath := testlib.ADMIN_HELM_CHART_PATH
+		output := helm.RenderTemplate(t, options, testlib.ADMIN_HELM_CHART_PATH, "release-name", []string{"templates/statefulset.yaml"})
+
+		for _, obj := range testlib.SplitAndRenderStatefulSet(t, output, 1) {
+			t.Logf("Inspecting admin statefulset: %s", obj.Name)
+			checkSpecVolumes(t, obj.Spec.Template.Spec.Volumes, options, helmChartPath)
+			checkSidecarContainers(t, obj.Spec.Template.Spec.Containers, options, helmChartPath)
+		}
+	})
+
+	t.Run("testDatabaseStatefulsetSidecars", func(t *testing.T) {
+		// Run RenderTemplate to render the template and inspect database statefulset
+		helmChartPath := testlib.DATABASE_HELM_CHART_PATH
+		output := helm.RenderTemplate(t, options, testlib.DATABASE_HELM_CHART_PATH, "release-name", []string{"templates/statefulset.yaml"})
+
+		for _, obj := range testlib.SplitAndRenderStatefulSet(t, output, 1) {
+			t.Logf("Inspecting database statefulset: %s", obj.Name)
+			checkSpecVolumes(t, obj.Spec.Template.Spec.Volumes, options, helmChartPath)
+			checkSidecarContainers(t, obj.Spec.Template.Spec.Containers, options, helmChartPath)
+		}
+	})
+
+	t.Run("testDatabaseDeploymentSidecars", func(t *testing.T) {
+		// Run RenderTemplate to render the template and inspect database statefulset
+		helmChartPath := testlib.DATABASE_HELM_CHART_PATH
+		output := helm.RenderTemplate(t, options, testlib.DATABASE_HELM_CHART_PATH, "release-name", []string{"templates/deployment.yaml"})
+
+		for _, obj := range testlib.SplitAndRenderDeployment(t, output, 1) {
+			t.Logf("Inspecting database deployment: %s", obj.Name)
+			checkSpecVolumes(t, obj.Spec.Template.Spec.Volumes, options, helmChartPath)
+			checkSidecarContainers(t, obj.Spec.Template.Spec.Containers, options, helmChartPath)
+		}
+	})
+
+	t.Run("testDatabaseDaemonsetSidecars", func(t *testing.T) {
+		// Run RenderTemplate to render the template and inspect database daemonset
+		options.SetValues["database.enableDaemonSet"] = "true"
+		helmChartPath := testlib.DATABASE_HELM_CHART_PATH
+		output := helm.RenderTemplate(t, options, testlib.DATABASE_HELM_CHART_PATH, "release-name", []string{"templates/daemonset.yaml"})
+
+		for _, obj := range testlib.SplitAndRenderDaemonSet(t, output, 1) {
+			t.Logf("Inspecting database daemonset: %s", obj.Name)
+			checkSpecVolumes(t, obj.Spec.Template.Spec.Volumes, options, helmChartPath)
+			checkSidecarContainers(t, obj.Spec.Template.Spec.Containers, options, helmChartPath)
+		}
+	})
+}
+
+func TestNuoDBCollectorSidecarsEnabled(t *testing.T) {
+
+	options := &helm.Options{
+		SetValues: map[string]string{
+			"nuocollector.enabled":            "true",
+			"nuocollector.image.registry":     "docker.io",
+			"nuocollector.image.repository":   "nuodb/nuocd",
+			"nuocollector.image.tag":          "1.0.0",
+			"nuocollector.watcher.registry":   "docker.io",
+			"nuocollector.watcher.repository": "kiwigrid/k8s-sidecar",
+			"nuocollector.watcher.tag":        "latest",
```

Comment:
This would be future work, but it might be good to put the logic to detect the config map inside of nuocollector itself rather that relying on this kiwigrid thing.